tests/draft2019-09/core/id.json

```diff
@@ -0,0 +1,138 @@
+[
+    {
+        "description": "Invalid use of fragments in location-independent $id",
+        "schema": {"$ref": "https://json-schema.org/draft/2019-09/schema"},
+        "tests": [
+            {
+                "description": "Identifier name",
+                "data": {
+                    "$ref": "#foo",
+                    "$defs": {
+                        "A": {
+                            "$id": "#foo",
+                            "type": "integer"
+                        }
+                    }
+                },
+                "valid": false
+            },
+            {
+                "description": "Identifier path",
+                "data": {
+                    "$ref": "#/a/b",
+                    "$defs": {
+                        "A": {
+                            "$id": "#/a/b",
+                            "type": "integer"
+                        }
+                    }
+                },
+                "valid": false
+            },
+            {
+                "description": "Identifier name with absolute URI",
+                "data": {
+                    "$ref": "http://localhost:1234/bar#foo",
+                    "$defs": {
+                        "A": {
+                            "$id": "http://localhost:1234/bar#foo",
+                            "type": "integer"
+                        }
+                    }
+                },
+                "valid": false
+            },
+            {
+                "description": "Identifier path with absolute URI",
+                "data": {
+                    "$ref": "http://localhost:1234/bar#/a/b",
+                    "$defs": {
+                        "A": {
+                            "$id": "http://localhost:1234/bar#/a/b",
+                            "type": "integer"
+                        }
+                    }
+                },
+                "valid": false
+            },
+            {
+                "description": "Identifier name with base URI change in subschema",
```

Comment:
And here too, the description seems, if I understand these correctly in the spec, to be a bit of a non-sequitur -- these are invalid because `$id` can't be just a fragment anymore (and here even in some nested subschema), not much to do with base URIs changing, no?

Reply:
I can clarify “identifier name”.

Reply:
Since this whole thing is in a "bad fragments" test group, "identifier name" is effectively an "identifier plain name". I still think it's clear if one looks at the test group description.

Reply:
What is your suggestion to improve this because I think it's good as it is. However, that doesn't mean others won't have the same confusion, so I'm willing to change it, but I can't think of what yet. Thoughts?

Reply:
"plain name" wouldn't be accurate because some of the fragments being tested here are json pointers, not the plain-name style.